service/src/main.rs

```diff
@@ -741,6 +751,31 @@ fn enclave_account<E: EnclaveBase>(enclave_api: &E) -> AccountId32 {
 
 // Alice plays the faucet and sends some funds to the account if balance is low
 fn ensure_account_has_funds(api: &mut Api<sr25519::Pair, WsRpcClient>, accountid: &AccountId32) {
+	let funding_amount = amount_to_be_funded(api, accountid);
+	if 0 < funding_amount {
+		bootstrap_funds_from_alice(api, accountid, funding_amount);
+	}
+}
+
+// Amount that needs to be funded for this account to have at least existential_deposit * MIN_FUND_INCREASE_FACTOR (1_000_000_000_000)
+fn amount_to_be_funded(api: &mut Api<sr25519::Pair, WsRpcClient>, accountid: &AccountId32) -> u128 {
+	// check account balance
+	let free = api.get_free_balance(accountid).unwrap();
+	info!("TEE's free balance = {:?}", free);
+
+	let existential_deposit = api.get_existential_deposit().unwrap();
+	info!("Existential deposit is = {:?}", existential_deposit);
+	let funding_amount = existential_deposit * MIN_FUND_INCREASE_FACTOR - free;
+	info!("Funding amount is = {:?}", funding_amount);
+	funding_amount
+}
+
+// Alice sends some funds to the account
+fn bootstrap_funds_from_alice(
```

Comment:
Also here - and I know it was this way before - we should probably improve by returning a `Result` and not using `unwraps`?

Reply:
This is right at start up, before the worker starts to do anything. As long as it clear where the error has happened I don't mind either method. Where would you do the error handling?

Reply:
The error handling would have to be done at least one level above.. I think it's more future proof to do it 'cleanly' with a `Result`. But I agree then - if we can live with the unwraps for now, we can also leave it as is (depending on whether Edith feels like writing some more code, or would rather have this PR over with quickly 😛 ).